{{cookiecutter.package_name}}/src/{{cookiecutter.package_name}}/tools/cli.py

```diff
@@ -6,136 +6,159 @@
-from functools import wraps
-from sys import exit
 
 
-PROJ_NAME = '{{cookiecutter.package_name}}'
-Location = namedtuple('Location', ['loc_proper', 'loc_sanitized'])
+PROJECT_NAME = '{{cookiecutter.package_name}}'
+Location = namedtuple('Location', ['proper', 'sanitized'])
+
+
+def sanitize(*locations: str) -> Generator[Location]:
+    """Processes locations into tuples of proper and sanitized names.
+
+    Sanitized location strings are all lower case, have spaces replaced
+    by underscores, and have apostrophes replaced by dashes.
+
+    Parameters
+    ----------
+    locations
+        The locations to process formatted as proper location names.
+        Proper location names should come from GBD location set 1, the
+        location reporting hierarchy wherever possible. If using sub-national
+        locations, they may be found in GBD location set 35, the model
+        results location hierarchy.
+
+    Yields
+    -------
+        Named tuples with both the proper and sanitized location names
+
+    Examples
+    --------
+    >>> sanitize("Nigeria")
+    Location(proper="Nigeria", sanitized="nigeria")
 
+    >>> sanitize("Burkina Faso")
+    Location(proper="Burkina Faso", sanitized="burkina_faso")
 
-def sanitize(loc: str) -> (bool, Location):
-    loc_proper = loc.strip()
-    if loc_proper.startswith('#'):
-        loc_proper = loc_sanitized = ''
-    else:
-        loc_sanitized = loc_proper
-        loc_sanitized = re.sub("[- ,.&']", '_', loc_sanitized).lower()
-    return bool(loc_proper), Location(loc_proper, loc_sanitized)
+    >>> sanitize("Cote d'Ivoire")
+    Location(proper="Cote d'Ivoire", sanitized="cote_d-ivoire")
 
+    """
+    # TODO: Check that they're in a call to get_location_id from
+    #    vivarium_gbd_access.gbd.  Not doing this now because it's not
+    #    specified as a proper dependency in the setup.py and would be a
+    #    bit of a pain to do now.
+    for location in locations:
+        proper = location.strip()
+        sanitized = re.sub("[- ,.&']", '_', proper).lower()
+        yield Location(proper, sanitized)
+
+
+def parse_locations(locations_file: Optional[str], single_location: Optional[str]) -> List[str]:
+    """Parses location inputs into a list of location strings.
+
+    If no arguments are provided, this will default to the ``locations.txt``
+    file located in the repository model_specifications directory.
+
+    Parameters
+    ----------
+    locations_file
+        Path to a file containing a list of locations to generate model
+        specifications for.
+    single_location
+        Optional single location to generate a model specification for.
+
+    Returns
+    -------
+        A list of location strings.
+
+    Raises
+    ------
+    ValueError
+        If both ``locations_file`` and ``single_loc`` are provided or if
+        a ``locations_file`` with no locations is provided.
 
-def get_sanitized_locations(single_loc: str, loc_file: Path) -> List[Location]:
```

Comment:
Given how much of our work revolves around working with files and data, I try to be super strict about keeping I/O separate from transformation code.  Parsing the locations file and sanitizing the location strings feels like two very separate operations to me. In particular, they feel like operations whose implementation might change independently.  Moving the comment exclusion out of sanitization and into the file parsing makes managing that kind of change much easier.